lib/item/item.py

```diff
@@ -2042,30 +2064,73 @@ def _log_build_text(self, value, caller, source=None, dest=None):
         return txt
 
 
+    def _get_rule(self, rule_entry):
+        def convert_entry(entry, to):
+            returnvalue = entry
+            if isinstance(returnvalue, str) and to != "str":
+                try:
+                    # try to get value from item
+                    item = self.get_absolutepath(entry.strip().replace("sh.", ""), "log_rules")
+                    returnvalue = _items_instance.return_item(item).property.value
+                except Exception as e:
+                    pass
+            if isinstance(returnvalue, (str, int)) and to == "num":
+                try:
+                    returnvalue = float(returnvalue)
+                except ValueError as e:
+                    returnvalue = None
+            elif isinstance(entry, list):
+                entry = [convert_entry(val, self._type) for val in entry]
+            elif isinstance(returnvalue, (float, int)) and to == "str":
+                returnvalue = str(returnvalue)
+            if returnvalue is None:
+                logger.warning(f"Given log_rules entry '{entry}' is invalid.")
+            return returnvalue
+
+        defaults = {'filter': [], 'exclude': [], 'lowlimit': None, 'highlimit': None}
+        types = {'filter': 'list', 'exclude': 'list', 'lowlimit': 'num', 'highlimit': 'num'}
+        entry =  self._log_rules.get(rule_entry, defaults.get(rule_entry))
+        if entry is not None and entry != []:
+            entry = convert_entry(entry, types.get(rule_entry) or self._type)
+
+        return entry
+
     def _log_on_change(self, value, caller, source=None, dest=None):
         """
         Write log, if Item has attribute log_change set
         :return:
         """
+
         if self._log_change_logger is not None:
-            filter_list = self._log_rules.get('filter', [])
+            filter_list = self._get_rule('filter')
+            exclude_list = self._get_rule('exclude')
+            if filter_list != [] and exclude_list != []:
+                logger.warning(f"Defining filter AND exclude does not work. "
+                               f"Ignoring exclude list {exclude_list} "
+                               f"Using filter: {filter_list}")
 
             if self._type == 'num':
-                low_limit =  self._log_rules.get('lowlimit', None)
+                low_limit =  self._get_rule('lowlimit')
                 if low_limit:
```

Comment:
Zwei Anmerkungen: 
1. ohne die explizite Abfrage auf `self._type == 'num'` könnte das durchaus auch für strings, bytes o.ä. ohne weitere Änderung gehen. list und dict gehen nicht, und None macht auch Probleme.
2. Die Abfrage `if low_limit` ist aus meiner Sicht problematisch, weil die if-Bedingung bei `low_limit == 0` nicht erfüllt wäre und nicht ausgewertet würde. Besser wäre `if low_limit is not None:`. Dito bei high_limit.

Reply:
Guter Hinweis, Zweiteres hab ich aktualisiert. Magst du Punkt 1 selbst ändern bzw. klären, warum danach abgefragt wird (so war das ja bereits im Code)

Reply:
@msinn weißt du, ob es einen bestimmten Grund hat, dass lowlimit und highlimit im Log auf type num beschränkt sind? "Vergleichsfähige" Typen wie str, bytes, bytearray (noch mehr) können ja ohne Codeänderung mit < und > verglichen werden und könnten hier auch verwendet werden...
Ich will nur nicht jetzt Änderungen einbringen, die ggf. aus gutem Grund nicht implementiert wurden ;)

Reply:
Ja, der Grund ist, dass byte und bytearray, etc. keine SmartHomeNG Item-Typen sind.